### PR TITLE
Add SNS Topic Subscription Not Setted query

### LINKED ISSUE #1224: Add NS Topic Subscription Not Setted query for CloudFormation
### Platform
CloudFormation

### Provider
AWS

### Description
Ensure appropriate subscribers to each SNS topic

### PR DESCRIPTION
**Pull request by fabioGoncalvesCx**
Added new query for AWS CloudFormation.

Provide a CloudFormation resource and a resource to ensure that appropriate subscribers are setted to each SNS topic
Closes #1224

### CODE REVIEW

File: assets/queries/cloudFormation/sns_topic_subscription_not_setted/metadata.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "SNS_Topic_Subscription_Not_Setted",
+  "queryName": "SNS Topic Subscription Not Setted",
+  "severity": "LOW",
+  "category": "Monitoring",
+  "descriptionText": "Ensure appropriate subscribers to each SNS topic",
+  "descriptionUrl": "https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sns-topic.html#cfn-sns-topic-subscription"
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/sns_topic_subscription_not_setted/metadata.json:8]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/sns_topic_subscription_not_setted/test/negative.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: ''
+Resources:
+  SnsTopic:
+    Type: 'AWS::SNS::Topic'
+    Properties:
+      Subscription:
+        - Endpoint: email@example.com
+          Protocol: email
+      TopicName: alarm-action
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/sns_topic_subscription_not_setted/test/negative.yaml:10]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/sns_topic_subscription_not_setted/test/positive.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: ''
+Resources:
+  SnsTopic:
+    Type: 'AWS::SNS::Topic'
+    Properties:
+      TopicName: alarm-action
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/sns_topic_subscription_not_setted/test/positive.yaml:7]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/sns_topic_subscription_not_setted/metadata.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "SNS_Topic_Subscription_Not_Setted",
+  "queryName": "SNS Topic Subscription Not Setted",
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/sns_topic_subscription_not_setted/metadata.json:3]: Change **directory name** and **id** to: `sns_topic_without_subscriber`
Change **query name** to: `SNS Topic Without Subscriber`

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
Requesting changes

**Comment by markmishaevcx** [comment]
Scan submitted to Checkmarx

**Comment by markmishaevcx** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 4 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**3 Medium**<br />Low&nbsp;**1 Low**<br />Info&nbsp;**0 Info**<br />
### Violation Summary
Medium&nbsp;**3** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049059&projectid=730)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[30](https://github.com/Checkmarx/kics/blob/feature/sns_topic_subscription_not_setted/cmd/builder/main.go#L30) |Medium|Path_Traversal|cmd/builder/main.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049059&projectid=730&pathid=3)
[52](https://github.com/Checkmarx/kics/blob/feature/sns_topic_subscription_not_setted/pkg/kics/service.go#L52) |Medium|Denial_Of_Service_Resource_Exhaustion|pkg/kics/service.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049059&projectid=730&pathid=1)
[81](https://github.com/Checkmarx/kics/blob/feature/sns_topic_subscription_not_setted/test/main_test.go#L81) |Medium|Denial_Of_Service_Resource_Exhaustion|test/main_test.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049059&projectid=730&pathid=2)

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**